backend/app/services/diagnosis_agent.py

```
@@ -0,0 +1,486 @@
+                if cat == 'POP' and 'precipitation_prob' not in weather_dict: weather_dict['precipitation_prob'] = val
+                if cat == 'REH' and 'humidity' not in weather_dict: weather_dict['humidity'] = val
+                if cat == 'WSD' and 'wind_speed' not in weather_dict: weather_dict['wind_speed'] = val
+            
+            w_str = f"{weather_dict.get('temperature', '-')}℃, {weather_dict.get('humidity', '-')}%, {weather_dict.get('wind_speed', '-')}m/s, {weather_dict.get('precipitation_prob', '-')}%"
+            weather_cache[region] = (now, weather_dict)
+            return {"weather_data": weather_dict}
+        return {"weather_data": {"temperature": "-", "humidity": "-", "wind_speed": "-", "precipitation_prob": "-"}}
+    except Exception as e:
+        return {"weather_data": {"temperature": "-", "humidity": "-", "wind_speed": "-", "precipitation_prob": "-"}}
+
+async def fetch_ncpms(state: DiagnosisState) -> dict:
```

Ruff (0.15.10)

[warning] Too many return statements (8 > 6)

(PLR0911)